LFortran

certik opened the pull request:
<!--
Thank you very much for putting in this recipe PR!

This repository is very active, so if you need help with
a PR or once it's ready for review, please let the right people know.
There are language-specific teams for reviewing recipes.

Currently available teams are:
- python `@conda-forge/help-python`
- python/c hybrid `@conda-forge/help-python-c`
- r `@conda-forge/help-r`
- java `@conda-forge/help-java`
- nodejs `@conda-forge/help-nodejs`
- c/c++ `@conda-forge/help-c-cpp`
- perl `@conda-forge/help-perl`
- Julia `@conda-forge/help-julia`

If your PR doesn't fall into those categories please contact
the full review team `@conda-forge/staged-recipes`.

Due to GitHub limitations first time contributors to conda-forge are unable
to ping these teams.
Consider asking on our [Gitter channel](https://gitter.im/conda-forge/conda-forge.github.io)
if your recipe isn't reviewed promptly.
-->

Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vend other packages
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your
      recipe (see
      [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos)
      for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/lfortran```) and found it was in an excellent condition.

certik commented:
It's a weird [error](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=56711&view=logs&jobId=10bda42a-d420-5c28-a43f-bc4e66405e5c) that I am getting:
```
Created temporary directory: /tmp/pip-req-tracker-pg_i1mz5
Created requirements tracker '/tmp/pip-req-tracker-pg_i1mz5'
Created temporary directory: /tmp/pip-install-qnzvf1ev
Processing $SRC_DIR
  Created temporary directory: /tmp/pip-req-build-_discgh3
  Added file://$SRC_DIR to build tracker '/tmp/pip-req-tracker-pg_i1mz5'
    Preparing wheel metadata: started
    Running command /home/conda/feedstock_root/build_artifacts/lfortran_1564094647832/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/bin/python /home/conda/feedstock_root/build_artifacts/lfortran_1564094647832/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/lib/python3.6/site-packages/pip/_vendor/pep517/_in_process.py prepare_metadata_for_build_wheel /tmp/tmpj0kogzaq
    Traceback (most recent call last):
      File "/home/conda/feedstock_root/build_artifacts/lfortran_1564094647832/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/lib/python3.6/site-packages/pip/_vendor/pep517/_in_process.py", line 207, in <module>
        main()
      File "/home/conda/feedstock_root/build_artifacts/lfortran_1564094647832/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/lib/python3.6/site-packages/pip/_vendor/pep517/_in_process.py", line 197, in main
        json_out['return_val'] = hook(**hook_input['kwargs'])
      File "/home/conda/feedstock_root/build_artifacts/lfortran_1564094647832/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/lib/python3.6/site-packages/pip/_vendor/pep517/_in_process.py", line 69, in prepare_metadata_for_build_wheel
        return hook(metadata_directory, config_settings)
      File "/home/conda/feedstock_root/build_artifacts/lfortran_1564094647832/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/lib/python3.6/site-packages/setuptools/build_meta.py", line 155, in prepare_metadata_for_build_wheel
        self.run_setup()
      File "/home/conda/feedstock_root/build_artifacts/lfortran_1564094647832/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/lib/python3.6/site-packages/setuptools/build_meta.py", line 234, in run_setup
        self).run_setup(setup_script=setup_script)
      File "/home/conda/feedstock_root/build_artifacts/lfortran_1564094647832/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/lib/python3.6/site-packages/setuptools/build_meta.py", line 141, in run_setup
        exec(compile(code, __file__, 'exec'), locals())
      File "setup.py", line 2, in <module>
        import skbuild
    ModuleNotFoundError: No module named 'skbuild'
```
It says it can't find `skbuild` which is in the package `scikit-build` which is listed in the build requirements in meta.yaml:
```yaml
requirements:
  build:
    - {{ compiler('c') }}
    - python
    - setuptools
    - scikit-build
    - cmake
    - ninja
```
and it gets installed, as seen from the log:
```
## Package Plan ##

  environment location: /home/conda/feedstock_root/build_artifacts/lfortran_1564094647832/_build_env


The following NEW packages will be INSTALLED:

    _libgcc_mutex:          0.1-main                   defaults   
...
    scikit-build:           0.9.0-py36hf484d3e_0       conda-forge
```
So I don't know why it would not be available.

certik commented:
Here is the contents of the binary package on linux:
```
├── bin
│   └── lfort
├── info
│   ├── about.json
│   ├── files
│   ├── git
│   ├── hash_input.json
│   ├── has_prefix
│   ├── index.json
│   ├── licenses
│   │   └── LICENSE
│   ├── paths.json
│   ├── recipe
│   │   ├── conda_build_config.yaml
│   │   ├── LICENSE
│   │   ├── meta.yaml
│   │   └── meta.yaml.template
│   ├── repodata_record.json
│   └── test
│       ├── run_test.py
│       ├── run_test.sh
│       └── test_time_dependencies.json
├── lib
│   └── python3.7
│       └── site-packages
│           ├── lfortran
│           │   ├── adapters
│           │   │   ├── gfortran
│           │   │   │   ├── __init__.py
│           │   │   │   ├── mod.py
│           │   │   │   └── __pycache__
│           │   │   │       ├── __init__.cpython-37.pyc
│           │   │   │       └── mod.cpython-37.pyc
│           │   │   ├── __init__.py
│           │   │   └── __pycache__
│           │   │       └── __init__.cpython-37.pyc
│           │   ├── asr
│           │   │   ├── asr_check.py
│           │   │   ├── asr.py
│           │   │   ├── asr_to_ast.py
│           │   │   ├── builder.py
│           │   │   ├── __init__.py
│           │   │   ├── pprint.py
│           │   │   ├── __pycache__
│           │   │   │   ├── asr_check.cpython-37.pyc
│           │   │   │   ├── asr.cpython-37.pyc
│           │   │   │   ├── asr_to_ast.cpython-37.pyc
│           │   │   │   ├── builder.cpython-37.pyc
│           │   │   │   ├── __init__.cpython-37.pyc
│           │   │   │   └── pprint.cpython-37.pyc
│           │   │   └── tests
│           │   │       ├── __init__.py
│           │   │       ├── __pycache__
│           │   │       │   ├── __init__.cpython-37.pyc
│           │   │       │   ├── test_builder.cpython-37.pyc
│           │   │       │   └── test_pprint.cpython-37.pyc
│           │   │       ├── test_builder.py
│           │   │       └── test_pprint.py
│           │   ├── ast
│           │   │   ├── ast.py
│           │   │   ├── ast_to_src.py
│           │   │   ├── __init__.py
│           │   │   ├── __pycache__
│           │   │   │   ├── ast.cpython-37.pyc
│           │   │   │   ├── ast_to_src.cpython-37.pyc
│           │   │   │   ├── __init__.cpython-37.pyc
│           │   │   │   └── utils.cpython-37.pyc
│           │   │   ├── tests
│           │   │   │   ├── case_sensitivity_results.py
│           │   │   │   ├── controlflow_results.py
│           │   │   │   ├── decl_results.py
│           │   │   │   ├── expr_results.py
│           │   │   │   ├── __init__.py
│           │   │   │   ├── interactive_results.py
│           │   │   │   ├── module_results.py
│           │   │   │   ├── program_results.py
│           │   │   │   ├── __pycache__
│           │   │   │   │   ├── case_sensitivity_results.cpython-37.pyc
│           │   │   │   │   ├── controlflow_results.cpython-37.pyc
│           │   │   │   │   ├── decl_results.cpython-37.pyc
│           │   │   │   │   ├── expr_results.cpython-37.pyc
│           │   │   │   │   ├── __init__.cpython-37.pyc
│           │   │   │   │   ├── interactive_results.cpython-37.pyc
│           │   │   │   │   ├── module_results.cpython-37.pyc
│           │   │   │   │   ├── program_results.cpython-37.pyc
│           │   │   │   │   ├── statements_results.cpython-37.pyc
│           │   │   │   │   ├── subroutine_results.cpython-37.pyc
│           │   │   │   │   ├── test_ast.cpython-37.pyc
│           │   │   │   │   ├── test_dump.cpython-37.pyc
│           │   │   │   │   ├── test_files.cpython-37.pyc
│           │   │   │   │   ├── test_parser.cpython-37.pyc
│           │   │   │   │   ├── test_utils.cpython-37.pyc
│           │   │   │   │   ├── test_visitor.cpython-37.pyc
│           │   │   │   │   └── use_results.cpython-37.pyc
│           │   │   │   ├── statements_results.py
│           │   │   │   ├── subroutine_results.py
│           │   │   │   ├── test_ast.py
│           │   │   │   ├── test_dump.py
│           │   │   │   ├── test_files.py
│           │   │   │   ├── test_parser.py
│           │   │   │   ├── test_utils.py
│           │   │   │   ├── test_visitor.py
│           │   │   │   └── use_results.py
│           │   │   └── utils.py
│           │   ├── cli.py
│           │   ├── codegen
│           │   │   ├── asr_to_llvm.py
│           │   │   ├── evaluator.py
│           │   │   ├── gen.py
│           │   │   ├── __init__.py
│           │   │   ├── __pycache__
│           │   │   │   ├── asr_to_llvm.cpython-37.pyc
│           │   │   │   ├── evaluator.cpython-37.pyc
│           │   │   │   ├── gen.cpython-37.pyc
│           │   │   │   └── __init__.cpython-37.pyc
│           │   │   └── tests
│           │   │       ├── __init__.py
│           │   │       ├── __pycache__
│           │   │       │   ├── __init__.cpython-37.pyc
│           │   │       │   ├── test_arrays.cpython-37.pyc
│           │   │       │   ├── test_asr_to_llvm.cpython-37.pyc
│           │   │       │   ├── test_do_loops.cpython-37.pyc
│           │   │       │   ├── test_expr.cpython-37.pyc
│           │   │       │   ├── test_fortran_eval.cpython-37.pyc
│           │   │       │   ├── test_intrinsics.cpython-37.pyc
│           │   │       │   ├── test_llvm_eval.cpython-37.pyc
│           │   │       │   └── test_symbol_scope.cpython-37.pyc
│           │   │       ├── test_arrays.py
│           │   │       ├── test_asr_to_llvm.py
│           │   │       ├── test_do_loops.py
│           │   │       ├── test_expr.py
│           │   │       ├── test_fortran_eval.py
│           │   │       ├── test_intrinsics.py
│           │   │       ├── test_llvm_eval.py
│           │   │       └── test_symbol_scope.py
│           │   ├── __init__.py
│           │   ├── parser
│           │   │   ├── fortranLexer.py
│           │   │   ├── fortranParser.py
│           │   │   ├── fortranVisitor.py
│           │   │   ├── __init__.py
│           │   │   ├── parser.py
│           │   │   └── __pycache__
│           │   │       ├── fortranLexer.cpython-37.pyc
│           │   │       ├── fortranParser.cpython-37.pyc
│           │   │       ├── fortranVisitor.cpython-37.pyc
│           │   │       ├── __init__.cpython-37.pyc
│           │   │       └── parser.cpython-37.pyc
│           │   ├── prompt.py
│           │   ├── __pycache__
│           │   │   ├── cli.cpython-37.pyc
│           │   │   ├── __init__.cpython-37.pyc
│           │   │   └── prompt.cpython-37.pyc
│           │   ├── semantic
│           │   │   ├── analyze.py
│           │   │   ├── ast_to_asr.py
│           │   │   ├── __init__.py
│           │   │   ├── kinds.py
│           │   │   ├── __pycache__
│           │   │   │   ├── analyze.cpython-37.pyc
│           │   │   │   ├── ast_to_asr.cpython-37.pyc
│           │   │   │   ├── __init__.cpython-37.pyc
│           │   │   │   └── kinds.cpython-37.pyc
│           │   │   └── tests
│           │   │       ├── __init__.py
│           │   │       ├── __pycache__
│           │   │       │   ├── __init__.cpython-37.pyc
│           │   │       │   ├── test_ast_to_asr.cpython-37.pyc
│           │   │       │   └── test_variables.cpython-37.pyc
│           │   │       ├── test_ast_to_asr.py
│           │   │       └── test_variables.py
│           │   └── tests
│           │       ├── __init__.py
│           │       ├── __pycache__
│           │       │   ├── __init__.cpython-37.pyc
│           │       │   ├── test_lld.cpython-37.pyc
│           │       │   └── utils.cpython-37.pyc
│           │       ├── test_lld.py
│           │       └── utils.py
│           └── lfortran-0.1.14.dist-info
│               ├── INSTALLER
│               ├── METADATA
│               ├── RECORD
│               ├── top_level.txt
│               └── WHEEL
└── share
    └── lfortran
        ├── lib
        │   ├── liblfortran.so
        │   └── liblfortran_static.a
        └── nb
            └── Demo.ipynb

39 directories, 153 files
```
So only the LFortran package is installed, nothing else.

isuruf commented on recipes/lfortran/LICENSE:
```diff
@@ -0,0 +1,17 @@
+Copyright 2019. Triad National Security, LLC. All rights reserved.
```
Can you add the `LICENSE` file to `MANIFEST.in` so that sdist will contain the license?

certik replied:
Yes. I have to do this upstream.

certik replied:
Fixed in https://gitlab.com/lfortran/lfortran/merge_requests/228/. I'll wait if there are more things to fix before making a new upstream release.

isuruf replied:
Can you also print `CMAKE_C_FLAGS` in addition to `CMAKE_C_FLAGS_RELEASE`?

isuruf replied:
Can you remove this file now?

certik replied:
Done.

isuruf commented on recipes/lfortran/meta.yaml:
```diff
@@ -0,0 +1,60 @@
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - python
```
Remove this. It's only need in `host`

isuruf commented:
For the osx failures, see https://github.com/scikit-build/scikit-build/issues/362#issuecomment-470150020

isuruf commented:
For the windows error, add `ninja` to build requirements.

isuruf commented:
@jcfr, do you know why skbuild doesn't try NMake or Ninja in windows?

certik commented:
> For the windows error, add `ninja` to build requirements.

`ninja` already is in build requirements.

isuruf commented on recipes/lfortran/build.sh:
```diff
@@ -0,0 +1,16 @@
+    python -m pip install . -vv \
+      --install-option="bdist_wheel" \
+      --install-option="--plat-name" \
+      --install-option="macosx-10.9-x86_64"
```
Use `${MACOSX_DEPLOYMENT_TARGET}` instead of 10.9

certik replied:
Good point. Do you understand what the issue is here? There might be an easier fix in the CMakeLists.txt than fiddling with things at the pip level. It seems there is a failure later on that perhaps could be avoided if CMake can use the proper `MACOSX_DEPLOYMENT_TARGET` on its own.

isuruf replied:
skbuild will send `-DCMAKE_OSX_DEPLOYMENT_TARGET=10.6` to cmake overriding the environment variable that CMake usually uses.

isuruf replied:
skbuild also ignores `CFLAGS, LDFLAGS` env variables as well, which is annoying.

certik replied:
It almost seems it's a bug in skbuild that we are trying to patch here. Since skbuild is sending the wrong `CMAKE_OSX_DEPLOYMENT_TARGET`.

certik commented:
Why is it trying to build wheels first? Or is that a normal procedure.

Windows on Appveyor works, Windows on azure fails with:
```
  CMake Error at CMakeLists.txt:2 (PROJECT):
    Failed to run MSBuild command:

      MSBuild.exe

    to get the value of VCTargetsPath:

      Microsoft (R) Build Engine version 4.8.3761.0
      [Microsoft .NET Framework, version 4.0.30319.42000]
      Copyright (C) Microsoft Corporation. All rights reserved.

      Build started 7/26/2019 12:48:17 AM.
      Project "C:\Users\VssAdministrator\AppData\Local\Temp\pip-req-build-kzujc_1d\_cmake_test_compile\build\CMakeFiles\3.14.5\VCTargetsPath.vcxproj" on node 1 (default targets).
      C:\Users\VssAdministrator\AppData\Local\Temp\pip-req-build-kzujc_1d\_cmake_test_compile\build\CMakeFiles\3.14.5\VCTargetsPath.vcxproj(15,2): error MSB4019: The imported project "C:\Microsoft.Cpp.Default.props" was not found. Confirm that the path in the <Import> declaration is correct, and that the file exists on disk.
      Done Building Project "C:\Users\VssAdministrator\AppData\Local\Temp\pip-req-build-kzujc_1d\_cmake_test_compile\build\CMakeFiles\3.14.5\VCTargetsPath.vcxproj" (default targets) -- FAILED.

      Build FAILED.

      "C:\Users\VssAdministrator\AppData\Local\Temp\pip-req-build-kzujc_1d\_cmake_test_compile\build\CMakeFiles\3.14.5\VCTargetsPath.vcxproj" (default target) (1) ->
        C:\Users\VssAdministrator\AppData\Local\Temp\pip-req-build-kzujc_1d\_cmake_test_compile\build\CMakeFiles\3.14.5\VCTargetsPath.vcxproj(15,2): error MSB4019: The imported project "C:\Microsoft.Cpp.Default.props" was not found. Confirm that the path in the <Import> declaration is correct, and that the file exists on disk.

          0 Warning(s)
          1 Error(s)

      Time Elapsed 00:00:00.44


    Exit code: 1
```
I wonder what this error means.

To simplify things, I only build for Python 3.7 for now.

The issue on macOS is:
```
...
    adding 'lfortran-0.1.14.dist-info/WHEEL'
    adding 'lfortran-0.1.14.dist-info/top_level.txt'
    adding 'lfortran-0.1.14.dist-info/RECORD'
    removing _skbuild/macosx-10.9-x86_64-3.7/setuptools/bdist.macosx-10.9-x86_64/wheel
  Running setup.py install for lfortran: finished with status 'done'
ERROR: Could not install packages due to an EnvironmentError.
Consider using the `--user` option or check the permissions.
Traceback (most recent call last):
  File "/usr/local/miniconda/conda-bld/lfortran_1564101787101/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_plac/lib/python3.7/site-packages/pip/_internal/commands/install.py", line 407, in run
    use_user_site=options.use_user_site,
  File "/usr/local/miniconda/conda-bld/lfortran_1564101787101/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_plac/lib/python3.7/site-packages/pip/_internal/req/__init__.py", line 58, in install_given_reqs
    **kwargs
  File "/usr/local/miniconda/conda-bld/lfortran_1564101787101/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_plac/lib/python3.7/site-packages/pip/_internal/req/req_install.py", line 999, in install
    ensure_dir(egg_info_dir)
  File "/usr/local/miniconda/conda-bld/lfortran_1564101787101/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_plac/lib/python3.7/site-packages/pip/_internal/utils/misc.py", line 145, in ensure_dir
    os.makedirs(path)
  File "/usr/local/miniconda/conda-bld/lfortran_1564101787101/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_plac/lib/python3.7/os.py", line 221, in makedirs
    mkdir(name, mode)
PermissionError: [Errno 13] Permission denied: '/lfortran-0.1.14-py3.7.egg-info'
```
That looks almost like an skbuild error.

certik commented:
I wonder if the issue with Windows is that skbuild requires Visual Studio 2015, which might not be properly installed on the image, since the [official documentation](https://github.com/microsoft/azure-pipelines-image-generation/blob/6e9998b612bfa243478f026c65ce079fe4b6940c/images/win/Vs2017-Server2016-Readme.md#visual-studio-2017-enterprise) only mentions Visual Studio 2017.
```
  -- Trying "Visual Studio 14 2015 Win64" generator - failure
  --------------------------------------------------------------------------------

  ********************************************************************************
  scikit-build could not get a working generator for your system. Aborting build.

  Building windows wheels for Python 3.7 requires Microsoft Visual Studio 2015.
  Get it with "Microsoft Visual C++ Build Tools":

    http://landinghub.visualstudio.com/visual-cpp-build-tools

  Or with "Visual Studio 2015":

    https://visualstudio.com/
```

isuruf commented:
On azure, msbuild is broken. But skbuild should have used the Ninja or NMake generator which works fine. For windows, if appveyor works, then that's fine. azure is still experimental.

certik commented:
I don't know how to force it to use ninja on azure. But if you say Appveyor is fine, then I'll let this one go.

The macOS failure needs to be investigated, not sure what is causing it.

certik commented:
Thanks @isuruf for your help. I think this is pretty close, just the macOS error needs to be fixed.

I can't believe how fragile the build is on all platforms. Do you have any tips how to make it more robust?

One option is to not use pip at all, remove it from pypi, and only use Conda, and install this using CMake only. So CMake would install the Python files also. The build script would just call cmake directly. And I know how to make CMake work on all platforms.

certik commented:
The build failed due to network error. Closed reopened to trigger a new build.

certik commented:
It failed with:
```
    usage: setup.py [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
       or: setup.py --help [cmd1 cmd2 ...]
       or: setup.py --help-commands
       or: setup.py cmd --help

    error: option -D not recognized
```

certik commented on recipes/lfortran/build.sh:
```diff
@@ -7,8 +7,7 @@ printenv
-      --install-option="build" \
-      --install-option="--" \
+      --install-option="build_ext" \
       --install-option="-DCMAKE_OSX_DEPLOYMENT_TARGET=${MACOSX_DEPLOYMENT_TARGET}"
```
Are you sure the previous patch didn't work? The build got canceled. I think you have to put in the `--`.

certik commented:
@isuruf it works! Thanks a lot for this. I can't believe how painful this was. I am going to port this patch upstream, so that it simplifies this build.

scopatz commented on recipes/lfortran/build.sh:
```diff
@@ -0,0 +1,13 @@
+    python -m pip install . -vv
+else
+    # Linux
+    python -m pip install . -vv
```
what is different in these bodies? and shouldn't it be $PYTHON?

certik replied:
It used to be different as we experimented with various approaches. Now it is the same, so I am going to fold this into the meta.yaml file itself.

certik replied:
Fixed in 223322eec8122540295d647230f1ff48d21b88bb.

certik commented:
I ported the `setup.py` patch upstream in https://gitlab.com/lfortran/lfortran/merge_requests/229. I will make a new release and simplify this PR further.

@scopatz, what should be done about the failing Windows build on Azure? @isuruf says we can probably ignore it, but it bothers me that it fails, but I don't know how to fix it with scikit-build.

isuruf commented:
> I don't know how to fix it with scikit-build.

Opened an upstream issue, https://github.com/scikit-build/scikit-build/issues/439

certik commented:
@isuruf you fixed it on Windows! Thank you.

certik commented:
I moved the old messy history here: https://github.com/certik/staged-recipes/commits/lfortran-old, and squashed the commits in this PR. After tests pass, this should be ready to go in.

certik commented:
Tests pass, this is ready to go in as far as I am concerned. 

@isuruf, @scopatz.

certik commented:
@isuruf I removed the LICENSE file. All tests pass.

certik commented:
Thanks @isuruf!
